knimin/tests/test_barcode_util.py

```diff
@@ -281,5 +284,239 @@ def test_build_email(self):
+        self.assertIn('Subprojects: []', response.body)
+        self.assertIn('Barcode: %s' % barcode, response.body)
+
+        # TODO: I think we have a problem with spelling the word 'received'!
+        barcode_status_received = '000000001'
+
+        # check that neighter option from the combobox is selected, if status
+        # is empty
+        barcode_status_empty = "000001003"
+        response = self.get('/barcode_util/',
+                            {'barcode': barcode_status_empty})
+        self.assertEqual(response.code, 200)
+        self.assertIn('<option value="Recieved">Recieved</option>',
+                      response.body)
+        self.assertIn('<option value=""></option>', response.body)
+
+        # check that neighter option from the combobox is selected, if status
+        # is an empty quote
+        barcode_status_quote = "000001420"
+        response = self.get('/barcode_util/',
+                            {'barcode': barcode_status_quote})
+        self.assertEqual(response.code, 200)
+        self.assertIn('<option value="Recieved">Recieved</option>',
+                      response.body)
+        self.assertIn('<option value="" selected></option>', response.body)
+
+        # test correct option is set for barcodes with biomass remainings
+        barcode_biomass_Y = "000003336"
+        response = self.get('/barcode_util/',
+                            {'barcode': barcode_biomass_Y})
+        self.assertEqual(response.code, 200)
+        self.assertIn('<option value="N" >No</option>', response.body)
+        self.assertIn('<option value="Y" selected>Yes</option>',
+                      response.body)
+        self.assertIn('<option value="Unknown" >Unknown</option>',
+                      response.body)
+
+        # test correct option is set for barcodes without biomass remainings
+        barcode_biomass_N = "000004244"
+        response = self.get('/barcode_util/',
+                            {'barcode': barcode_biomass_N})
+        self.assertEqual(response.code, 200)
+        self.assertIn('<option value="N" selected>No</option>', response.body)
+        self.assertIn('<option value="Y" >Yes</option>', response.body)
+        self.assertIn('<option value="Unknown" >Unknown</option>',
+                      response.body)
+
+        # test correct option is set for barcodes without info about biomass
+        # remainings
+        barcode_biomass_empty = "000004242"
+        response = self.get('/barcode_util/',
+                            {'barcode': barcode_biomass_empty})
+        self.assertEqual(response.code, 200)
+        self.assertIn('<option value="N" >No</option>', response.body)
+        self.assertIn('<option value="Y" >Yes</option>', response.body)
+        self.assertIn('<option value="Unknown" selected>Unknown</option>',
+                      response.body)
+
+        # check if sequencing status is set to ''
+        # (TODO: currently it is set to WAITING, which seems to be wrong!)
```

Comment:
We have the following values in our DB for the column sequencing_status:
"''"
""
"FAILED_SEQUENCING"
"FAILED_SEQUENCING_1"
"FAILED_SEQUENCING_2"
"SUCCESS"
"WAITING"
Thus, I figure there should be a difference between a barcode marked as "waiting" and a barcode without any information, especially since there is the missing value option in the combo box `<option value="" ></option>`